Code/Framework/AzCore/AzCore/Metrics/EventLoggerReflectUtils.cpp

```diff
@@ -462,7 +462,8 @@ namespace AZ::Metrics
                     eventPhaseArgs.VisitArgs(AppendArgFields);
                     asyncArgs.m_args = eventFields;
 
-                    asyncArgs.m_id = eventPhaseArgs.m_id.value_or(AZStd::string_view{});
+                    auto tmp = eventPhaseArgs.m_id.value_or(AZStd::string_view{});
+                    asyncArgs.m_id = tmp;
```

Comment:
This can be fixed by checking if the optional has a value before assigning it to the `AsyncArgs` class ID `string_view` member
```suggestion
                    if (eventPhaseArgs.m_id.has_value())
                    {
                        asyncArgs.m_id = eventPhaseArgs.m_id.value();
                    }
```